flocker/node/agents/blockdevice.py

```diff
@@ -1121,8 +1121,10 @@ def list_live_nodes():
         This is used to figure out which nodes are dead, so that other
         nodes can do the detach.
 
-        :returns: A collection of ``unicode`` compute instance IDs, compatible
-            with those returned by ``IBlockDeviceAPI.compute_instance_id``.
+        :returns: A mapping of ``unicode`` compute instance IDs
```

Comment:
Thought about this more... I think we might want to introduce a new type here rather than returning a raw mapping.

Something like returning a list of `CloudComputeInstance`'s:

```
class CloudComputeInstance(PClass):
   id = field(type=unicode)
   ip_addresses = pvector_field(type=unicode)
```

Rational:

I don't know what the following does without looking at the source:
```
"ssh root@{}".format(api.list_live_nodes()['i-3ab8f212'][0])
```

I can guess what the following does without going to the source:
```
"ssh root@{}"format(next(x for x in api.list_live_nodes() if x.compute_id = 'i-3ab8f212').ip_addresses[0])
```